numpyro/infer/mixed_hmc.py

```diff
@@ -0,0 +1,211 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+from functools import partial
+
+from jax import jacfwd, grad, lax, ops, random
+import jax.numpy as jnp
+
+from numpyro.infer.hmc import momentum_generator
+from numpyro.infer.hmc_gibbs import DiscreteHMCGibbs, HMCGibbsState
+from numpyro.infer.hmc_util import euclidean_kinetic_energy, warmup_adapter
+from numpyro.util import cond, fori_loop, identity, ravel_pytree
+
+
+class MixedHMC(DiscreteHMCGibbs):
+    """
+    Implementation of Mixed Hamiltonian Monte Carlo (reference [1]).
+
+    .. note:: The number of discrete sites to update each time (`n_D` in reference [1])
```

Comment:
nit: The number of discrete sites to update at each MCMC iteration (`n_D` in reference [1]) is fixed at 1